### ISSUE TITLE
RCS1231 is triggered for CancellationToken in non-async method

### ISSUE THREAD

**Issue by sliekens** [JosefPihrt/Roslynator#1702]
There is a discrepancy in the RCS1231 analyzer between sync and async methods

> RCS1231: Make parameter ref read-only
> https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1231/

``` csharp
internal sealed class Rcs1231
{
    // Async method => no warning
    public async Task DoThisWay(CancellationToken cancellationToken)
    {
        await Task.FromCanceled(cancellationToken).ConfigureAwait(false);
    }

    // Sync method returning Task => RCS1231 warning on CancellationToken parameter
    public Task DoThatWay(CancellationToken cancellationToken)
    {
        return Task.FromCanceled(cancellationToken);
    }
}
```
